internal/provider/resource_group_members.go

```diff
@@ -213,12 +212,18 @@ func resourceGroupMembersRead(ctx context.Context, d *schema.ResourceData, meta
 
 	members := make([]interface{}, len(membersObj.Members))
 	for i, member := range membersObj.Members {
+		// Use value if presnet or default as "delivery_settings" is not provided by API
+		deliverySettings := resourceGroupMembers().Schema["members"].Elem.(*schema.Resource).Schema["delivery_settings"].Default.(string)
+		if ds := d.Get(fmt.Sprintf("members.%d.delivery_settings", i)); ds != nil {
```

Comment:
I realized that I copied this over from when I was testing it as a `TypeList`, so the index `i` here needs to actually be the hash of the object that we saved in the `Set` function. Then, as I was looking into that, I conferred with some other team members and we realized that example I sent you was actually broken there, broken in the sense that it actually ends up as the default Set function. Since I don't work with `TypeSet` much (or ever) I've spent the day going over this and am still trying to decide if we want a Set or a List. I'm sorry I don't have more for you today, but I'm running through tests and will hopefully have more details for you tomorrow! Thanks for your patience!

Reply:
Ok! I think we will have to keep it `TypeSet`. There's a bug in the diffSuppressFunc that will prevent us from moving forward with a list. That being said...a coworker and I discussed it and rather than trying to find out the hash here, we can do a `configMembers := d.Get("members").(*schema.Set)`, then loop through there to find the correct configured member and just grab it's `deliverySettings`. Does that make sense?

```
configMembers := d.Get("members").(*schema.Set)

	members := make([]interface{}, len(membersObj.Members))
	for i, member := range membersObj.Members {

		// Use value if present or default as "delivery_settings" is not provided by API
		deliverySettings := deliverySettingsDefault

		for _, cm := range configMembers.List() {
			cMem := cm.(map[string]interface{})
			if cMem["email"].(string) == member.Email {
				deliverySettings = cMem["delivery_settings"].(string)
				break
			}
		}
```

You can also get rid of that `Set` function that I suggested you create earlier too. So sorry about all the confusion. And also, if you wouldn't mind making the `"ALL_MAIL"` a constant at the top (I have `deliverySettingsDefault` here) so we can have it defined in one place, and not have that long line of code we had earlier.

Again, sorry for all the confusion on this! If at any point you don't have the time to finish it off, let me know and I can pick it up! There may need to be some further tweaking in the tests as they seem a bit flakey,  but might just be some eventual consistency problems. Thanks so much for your help on this!

Reply:
Hi Thanks for all the hints. I applied your suggestions, but if you have time to work on it, it would be great because I think we get the feature then quite faster ;)